Source/Scene/Model.js

```diff
@@ -1516,6 +1538,123 @@ define([
+                if (semantic === attributeSemantic) {
+                    for (var attributeVarName in technique.attributes) {
+                        if (technique.attributes.hasOwnProperty(attributeVarName)) {
+                            var name = technique.attributes[attributeVarName];
+                            if (name === parameter) {
+                                return attributeVarName;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+    
+    function modifyShaderForQuantizedAttributes(shader, programName, model, context) {
+        var gltf = model.gltf;
+        var quantizedUniforms = {};
+        model._quantizedUniforms[programName] = quantizedUniforms;
+
+        for (var meshId in gltf.meshes) {
+            if (gltf.meshes.hasOwnProperty(meshId)) {
+                var primitives = gltf.meshes[meshId].primitives;
+                for (var i = 0; i < primitives.length; i++) {
+                    var primitive = primitives[i];
+                    if (getProgramForPrimitive(model, primitive) === programName) {
+                        for (var attributeSemantic in primitive.attributes) {
+                            if (primitive.attributes.hasOwnProperty(attributeSemantic)) {
+                                var decodeUniformVarName = 'czm_u_dec_' + attributeSemantic.toLowerCase();
+                                if (!defined(quantizedUniforms[decodeUniformVarName])) {
+                                    var accessorId = primitive.attributes[attributeSemantic];
+                                    var quantizedAttributes = getQuantizedAttributes(model, accessorId);
+                                    if (defined(quantizedAttributes)) {
+                                        var attributeVarName = getAttributeVariableName(model, primitive, attributeSemantic);
+                                        var decodeMatrix = quantizedAttributes.decodeMatrix;
+                                        var newMain = 'czm_decoded_' + attributeSemantic;
+                                        var decodedAttributeVarName = attributeVarName.replace('a_', 'czm_a_dec_');
+                                        var size = Math.floor(Math.sqrt(decodeMatrix.length));
+
+                                        // replace usages of the original attribute with the decoded version, but not the declaration
+                                        shader = replaceAllButFirstInString(shader, attributeVarName, decodedAttributeVarName);
+                                        // declare decoded attribute
+                                        shader = 'vec' + (size - 1) + ' ' + decodedAttributeVarName + ';\n' + shader;
+                                        // splice decode function into the shader
+                                        var decode = '';
+                                        if (size === 5) {
```

Comment:
Add a comment that the decode transform is multiplied with the model matrix using 32-bit floating-point.

So if we run into an precision issue later, the potential culprit is more obvious.